ksqldb-rest-app/src/main/java/io/confluent/ksql/logging/query/QueryAnonymizingRewritePolicy.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.logging.query;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+
+import com.google.common.collect.ImmutableMap;
+import io.confluent.ksql.parser.ParsingException;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.QueryAnonymizer;
+import io.confluent.ksql.util.QueryGuid;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.apache.log4j.Logger;
+import org.apache.log4j.rewrite.RewritePolicy;
+import org.apache.log4j.spi.LoggingEvent;
+
+public final class QueryAnonymizingRewritePolicy implements RewritePolicy {
+  private static final QueryAnonymizer anonymizer = new QueryAnonymizer();
+  private final String namespace;
+  private final Boolean anonymizeQueries;
+
+  public QueryAnonymizingRewritePolicy(final KsqlConfig config) {
+    final String clusterNamespace =
+        config.getString(KsqlConfig.KSQL_CCLOUD_QUERYANONYMIZER_CLUSTER_NAMESPACE);
+    this.namespace =
+        clusterNamespace.isEmpty()
+            ? config.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG)
+            : clusterNamespace;
+    this.anonymizeQueries = config.getBoolean(KsqlConfig.KSQL_CCLOUD_QUERYANONYMIZER_ENABLED);
+  }
+
+  private QueryGuid buildGuids(final String query, final String anonymizedQuery) {
+    return new QueryGuid(namespace, query, anonymizedQuery);
+  }
+
+  @Override
+  public LoggingEvent rewrite(final LoggingEvent source) {
+    final ImmutableMap<String, String> oldMessage =
+        (ImmutableMap<String, String>) source.getMessage();
+    final String query = oldMessage.get("query");
+
+    final Map props = new HashMap(source.getProperties());
+
+    String anonymizedQuery;
+    try {
+      anonymizedQuery = anonymizer.anonymize(query);
+    } catch (ParsingException e) {
+      anonymizedQuery = "";
```

Comment:
Hmm I was trying to figure out the best semantic here. We certainly don't want to put the raw query content of things that failed to parse in here, because we might break our promise of not leaking PII. Perhaps it would make sense to add a log statement here? Otherwise I think silently failing should be good enough